app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -81,22 +90,28 @@ fun MainScreen() {
                     HomeRoute(
                         onNavigateToMovieDetails = { route ->
                             navController.navigate(route)
-                        }
+                        },
+                        homeViewModel
                     )
                 }
                 composable(NavigationItem.FavoritesDestination.route) {
                     FavoritesRoute(
                         onNavigateToMovieDetails = { route ->
                             navController.navigate(route)
                         },
-                        onClickLikeButton = { }
+                        favoritesViewModel
                     )
                 }
                 composable(
                     route = MovieDetailsDestination.route,
                     arguments = listOf(navArgument(MOVIE_ID_KEY) { type = NavType.IntType }),
-                ) {
-                    MovieDetailsRoute()
+                ) { navBackStackEntry ->
+                    var id: Int? = navBackStackEntry.arguments?.getInt(MOVIE_ID_KEY)
+                    if (id == null) id = 0
```

Comment:
val id = navBackStackEntry.arguments?.getInt(MOVIE_ID_KEY) ?: throw ... (If we are not passing id we want to crash the app because that mean we didn't implement something correctly and want to know it as soon as possible.)